lib/cache/expirations/factory.ts

```diff
@@ -0,0 +1,26 @@
+import { Factory } from '../../interfaces/factory';
+import { AbsoluteExpiration } from './AbsoluteExpiration';
+import { Expiration } from './Expiration';
+import { SlidingExpiration } from './SlidingExpiration';
+
+export class ExpirationFactory implements Factory<Expiration> {
+
+  constructor(
+    private readonly timeout: number,
+    private readonly expiration: 'absolute' | 'sliding',
+  ) { }
+
+  public create(): Expiration {
```

Comment:
Same as for storage, `expiration` is recommended to be part of the `create` method.